src/main/java/com/ibm/cloudant/kafka/connect/CustomValidator.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2022 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.ibm.cloudant.kafka.connect;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.ibm.cloudant.kafka.common.InterfaceConst;
+
+import org.apache.kafka.common.config.Config;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigValue;
+import org.apache.kafka.common.config.types.Password;
+
+// custom validations, for complex validation rules across multiple fields
+public class CustomValidator {
+
+    private Map<String, ConfigValue> values;
+    private List<ConfigValue> validations;
+
+    public CustomValidator(Map<String, String> props, ConfigDef config) {
+
+        validations = config.validate(props);
+        values = validations.stream().collect(Collectors.toMap(ConfigValue::name, Function.identity()));
+    }
+
+    public Config validate() {
+        validateBasicAuth();
+        return new Config(validations);
+    }
+
+    private void validateBasicAuth() {
+        String basic = "basic";
+        if (values.get(InterfaceConst.AUTH_TYPE).value().equals(basic)) {
```

Comment:
I guess as there is a default value this can't NPE, but I think in general safter to do the check the other way around and probably to ignore case, unless we're going to do some kind of case normalization.

```suggestion
        if (basic.equalsIgnoreCase(values.get(InterfaceConst.AUTH_TYPE).value()) {
```